File: src/test/resources/org/jetbrains/research/ml/dataset/anonymizer/transformation/java/anonymization/data/classes_and_methods/in_4_overriding_chain.java
public abstract class Test4_0  {
    abstract public void test(int x, int y);
}

public class Test4_1  extends Test4_0  {
    @Override
    public void test(int x, int y) {

    }
}

public class Test4_2 extends Test4_1  {
    @Override
    public void test(int x, int y) {

    }

    abstract public void test1(int x, int y);
}

public class Test4_3 extends Test4_2 {
    @Override
    public void test(int x, int y) {

    }

    @Override
    public void test1(int x, int y) {

    }
}

File: src/test/resources/org/jetbrains/research/ml/dataset/anonymizer/transformations/anonymization/kotlin/data/classes_and_methods/in_4_overriding_chain.kt
abstract class Test4_0 {
    abstract fun test(x: Int, y: Int)
}

open class Test4_1 : Test4_0() {
    override fun test(x: Int, y: Int) {}
}

abstract class Test4_2 : Test4_1() {
    override fun test(x: Int, y: Int) {}
    abstract fun test1(x: Int, y: Int)
}

class Test4_3 : Test4_2() {
    override fun test(x: Int, y: Int) {}
    override fun test1(x: Int, y: Int) {}
}
